x-pack/plugins/apm/server/index.ts

```diff
@@ -33,6 +36,9 @@ export function mergeConfigs(apmOssConfig: APMOSSConfig, apmConfig: APMXPackConf
     'xpack.apm.ui.maxTraceItems': apmConfig['ui.maxTraceItems'],
     'xpack.apm.ui.transactionGroupBucketSize': apmConfig['ui.transactionGroupBucketSize'],
     'xpack.apm.autocreateApmIndexPattern': apmConfig.autocreateApmIndexPattern,
+    'xpack.apm.serviceMapIndexPattern': apmConfig.serviceMapIndexPattern,
```

Comment:
`serviceMapIndexPattern` is misleading. It's not an index pattern (in the Kibana sense). This should follow the names of the other keys: `serviceMapIndices`.
Even though this is an xpack feature I'm contemplating whether it'll make more sense to keep this in `apm_oss` where all other indices are configured.

Reply:
I don't think it belongs in the apm_oss set of configs, since it's a premium feature. it will not be accessed if the use doesn't pass a license check.

Reply:
Yeah, I know, but it feels weird as an xpack user that you have to configure your indices differently. Let's get a second opinion on this.

Reply:
@bmorelli25 i'm interested in your take on this. Currently all of the apm indicies are namespaced under `apm.oss`. Examples: `apm_oss.errorIndices`, `apm_oss.transactionIndices` and `apm_oss.spanIndices`. Now we are also going to add a setting for Service Maps indices. 

From a docs perspective what do you think makes more sense to: 
 - `apm_oss.serviceMapIndices` (consistency with other index configuration option names)
 - `xpack.apm.serviceMapIndices` (Service Maps are only available in basic so is not used for anything in oss)

Reply:
>(Service Maps are only available in basic...)
should only be available in platinum

Reply:
My vote is for `xpack.apm.serviceMapIndices`.

> it feels weird as an xpack user that you have to configure your indices differently.

I agree, it does feel a little weird, but I think it's less weird than namespacing a platinum feature under the `oss` prefix.

Reply:
Makes sense. Then let's stick to `xpack.apm.serviceMapIndices`.